### PR TITLE
MQTT Autodiscovery for Homeassistant + added Battery Percentage and more

### PR DESCRIPTION
**Pull request by nxreigle**
Fixed the Autodiscovery for MQTT in HASS -> PubSubClient buffer was too small to fit the conf messages + some other misc problems.

- Added a switch in the Sensor firmware to send battery percentages Lora Packet:"NewMailCode,Batterypercentage"
- Gateway parses packet and sends status via mqtt
- Added Manufacturer + Model 

I kept it as a "door", just like the original contributer. To "empty" the mailbox, simply manually set the state to the string "empty" via mqtt payload through automation. 

Basically plug and play 😊

Screenshot 2023-09-05 at 23 27 37
Screenshot 2023-09-05 at 23 27 23

### CODE REVIEW

**Comment by PricelessToolkit** [comment]
Wow, thank you very mutch!!!!!!!!!!!!!!!!!!

**Comment by PricelessToolkit** [comment]
I separated sensor firmware "Mailbox_Guard_Sensor_MQTT_Auto_Discovery.ino" and old one "Mailbox_Guard_Sensor.ino"

**Comment by nxreigle** [comment]
Perfect 😊 I was also wondering about the most elegant solution to this. Separating them is probably the cleanest - taking in consideration we might find some more features to implement.

**Comment by PricelessToolkit** [comment]
Can you write me through pricelesstoolkit@gmail.com i have something to say :)

**Comment by nxreigle** [comment]
You've got Mail 😊

**Comment by PricelessToolkit** [comment]
I need to know your physical address 📦

**Comment by nxreigle** [comment]
Thanks a lot! :-) I've just sent another mail through iCloud - Sometimes googlemail is a bit picky when it comes to smaller, selfhosted mailservers. 

I'm really excited! Thanks!
